### PR TITLE
Gradle test configuration improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -197,6 +197,8 @@ subprojects {
             }
 
             register<JvmTestSuite>("funTest") {
+                configurations["funTestImplementation"].extendsFrom(configurations["testImplementation"])
```
Comment (fviernau) [inline build.gradle.kts:207]: Is this inheritance in terms of dependencies, e.g. funTestImpl inherits deps of testImpl?

Comment (sschuberth) [reply build.gradle.kts:207]: [Yes](https://docs.gradle.org/current/userguide/declaring_dependencies.html#sub:config-inheritance-composition).

Comment (fviernau) [reply build.gradle.kts:207]: Hm, I was unaware of that inheritance. I believe that both, `´test` and `funTest` do have dependencies which should not be shared. 

Would it make sense to share code by having 3 projects like:

1. common-test-utils
2. test-utils
3. fun-test-utils

Comment (sschuberth) [reply build.gradle.kts]: I've found a yet better way to share (only the basic) dependencies across test suites, please have a look.

**Pull request merged**